win32ss/user/user32/windows/cursoricon.c

```diff
@@ -2038,19 +2038,63 @@ HANDLE WINAPI CopyImage(
         return NULL;
     }
 
+    /* If hImage is NULL, then do fast return after setting LastError */
+    if (!hImage)
+    {
+        switch (uType)
+        {
+            case IMAGE_BITMAP:
+                SetLastError(ERROR_INVALID_HANDLE);
+                break;
+            case IMAGE_CURSOR:
+            case IMAGE_ICON:
+                SetLastError(ERROR_INVALID_CURSOR_HANDLE);
+                break;
+            default:
+                SetLastError(ERROR_INVALID_PARAMETER);
+        }
+        return NULL;
+    }
+
     switch(uType)
     {
         case IMAGE_BITMAP:
             return BITMAP_CopyImage(hImage, cxDesired, cyDesired, fuFlags);
```

Comment:
```suggestion
            if (!hImage)
            {
                SetLastError(ERROR_INVALID_HANDLE);
                break;
            }
            return BITMAP_CopyImage(hImage, cxDesired, cyDesired, fuFlags);
```
(seeing this I'm also wondering whether the NULL-check + last-error setting could be included within the BITMAP_CopyImage function itself.)

Reply:
Changed. Thanks.